index.js

```diff
@@ -349,10 +349,12 @@ class Raspi extends EventEmitter {
 
   digitalWrite(pin, value) {
     const pinInstance = this[getPinInstance](this.normalize(pin));
-    if (pinInstance.mode != OUTPUT_MODE) {
+    if (pinInstance.mode === INPUT_MODE && value === HIGH) {
+      pinInstance.peripheral = new DigitalInput({ pin: this.normalize(pin), pullResistor: PULL_DOWN });
```

Comment:
Do you think we could refactor pinMode a bit so that it can do this logic? By setting the peripheral manually here, we're skipping a few checks, namely if the pin supports GPIO (not all do). We could duplicate the logic here, but I'd prefer not to do that (DRY and all).

I think that the best way to do this would probably be to create a private `pinMode` method that takes one argument, `options`, an object. The public `pinMode` would maintain its existing signature and throw the number inside an object. That way, this line could call the private pinMode method with the full object here.

Does that make sense?